lib/SPIRVProducerPass.cpp

```diff
@@ -3707,8 +3768,18 @@ SPIRVProducerPassImpl::GenerateImageInstruction(CallInst *Call,
 
       Ops << result_type << Image << Coordinate;
       uint32_t mask = GetExtendMask(Call->getType(), is_int_image);
-      if (mask != 0)
+
+      bool isCoherent = isCoherentResource(Call->getArgOperand(0));
+
+      if (clspv::Option::VulkanMemoryModel() && isCoherent) {
```

Comment:
I don't see any coverage of this in the tests. In fact, I'm not sure clspv can generate a coherent image right now. That is likely a bug since where this functionality wasn't updated for read_write images. It would need a change in descriptor allocation. Please either fix that or file an issue to address it.

Separately, this could be tested using IR only. Take a look at the tests in Spv1p4 for how to run SPIRVProducer only tests.